src/nccl_ofi_net.c

```diff
@@ -1050,49 +1068,120 @@ static inline ncclResult_t nccl_ofi_progress(nccl_ofi_t *nccl_ofi_comp)
 }
 
 /*
- * @brief	Checks if the underlying hardware type matches the input platform type.
+ * @brief	Provides EC2 platform type as reported by the
+ * 		first line of
+ *		/sys/devices/virtual/dmi/id/product_name
  *
- * @return	-1, on error
- *		1, if it matches
- *		0, if it does not match
+ * @input	Buffer to store platform type, can be reallocated
+ * 		if required
+ *
+ * @return	-1, on file operation error
+ * 		0, on successful read
  */
-static int platform_type_matches(char *platform_type)
+static int get_platform_type(char **platform_type)
 {
 	char file[] = "/sys/devices/virtual/dmi/id/product_name";
 	FILE *fd = NULL;
-	int len = strlen(platform_type);
-	int idx = 0, ret = -1;
 	char ch;
+	int len = 0;
 
 	fd = fopen(file, "r");
 	if (fd == NULL) {
 		NCCL_OFI_WARN("Error opening file: %s", file);
 		return -1;
 	}
 
-	/* Read first line of the file */
+	/* Read first line of the file, reallocing the buffer as necessary */
 	while ((feof(fd) == 0) && (ferror(fd) == 0) && ((ch = fgetc(fd)) != '\n')) {
```

Comment:
Could this not just be an `fscanf` with a max length?

Reply:
I'm OK with shipping this, but I'll provide the other side of the argument.

I think it's extremely unlikely that a machine model is longer than 128 bytes.  If it happens to be, there's likely another bug somewhere and it would be better to simply error out the file read rather than allocating potentially unbounded memory.  

And as written the code actually encourages a much more likely programming error: it requires a buffer that has been `malloc`ed to be passed in.  Someone could easily pass in a stack or global buffer.  Usually the functions that return strings in this style do the `malloc` internally and then require the user to free the buffer.  Here we are mixing things up: `malloc` is on the outside, but `realloc` is on the inside.

Basically we traded a very unlikely footgun for one that is very likely to fire :)